lib/tokenize.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const split = String.prototype.split;
+const numbers = new Set(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']);
+
+const DEFAULT = Symbol('DEFAULT');
+const IN_SQUARE_BRACKETS = Symbol('IN_SQUARE_BRACKETS');
+const IMMEDIATELY_AFTER_SQUARE_BRACKETS = Symbol('IMMEDIATELY_AFTER_SQUARE_BRACKETS');
+
+exports.tokenize = function tokenize(str) {
```

Comment:
Can we keep the name as `stringToParts()`? Admittedly `tokenize()` is probably better. But I introduced some code that exports `stringToParts()`, so changing the name would be backwards breaking :(

Reply:
Is an alias enough?

Reply:
Done